packages/aws-cdk-lib/aws-cloudfront/lib/function.ts

```diff
@@ -222,11 +222,13 @@ export class Function extends Resource implements IFunction {
   }
 
   private generateName(): string {
-    const name = Stack.of(this).region + Names.uniqueId(this);
-    if (name.length > 64) {
-      return name.substring(0, 32) + name.substring(name.length - 32);
+    const maxNameLength = 64 - '${Token[AWS.Region.00]}'.length;
+    if (Names.uniqueId(this).length < maxNameLength) {
```

Comment:
```suggestion
    if (Names.uniqueId(this).length <= maxNameLength) {
```
Off by one I believe. Users are fine if their name came out to be this length as the non determinism comes when the token length plus the name is greater than 64.